sklearn/tree/tests/test_tree.py

```diff
@@ -1948,3 +1964,24 @@ def check_apply_path_readonly(name):
 @pytest.mark.parametrize("name", ALL_TREES)
 def test_apply_path_readonly_all_trees(name):
     check_apply_path_readonly(name)
+
+
+@pytest.mark.parametrize("criterion", ["mse", "poisson"])
+@pytest.mark.parametrize("tree", REG_TREES.values())
+def test_balance_property(criterion, tree):
+    """Test that sum(y_pred)=sum(y_true) on training set."""
+    # Choose a training set with non-negative targets
+    X, y = diabetes.data, diabetes.target
+    reg = tree(criterion=criterion)
+    reg.fit(X, y)
+    assert np.sum(reg.predict(X)) == pytest.approx(np.sum(y))
+
+
+def test_poisson_zero_nodes():
+    """Test that sum(y)=0 and therefore y_pred=0 never happens on nodes."""
```

Comment:
```suggestion
    """Test that sum(y)=0 and therefore y_pred=0 is forbidden on nodes."""
```

I was surprised this wasn't actually happening considering the dataset and then I saw your comment about this being enforced

Reply:
also nit but our convention for tests is to use comments instead of docstrings (don't ask me why ;) )